src/Azure/Orleans.Streaming.EventHubs/Providers/EventDataGeneratorStreamProvider/EventHubPartitionDataGenerator.cs

```diff
@@ -54,22 +54,25 @@ public bool TryReadEvents(int maxCount, out IEnumerable<EventData> events)
             while (count-- > 0)
             {
                 this.SequenceNumberCounter.Increment();
-                var eventData = EventHubBatchContainer.ToEventData<int>(this.serializationManager, this.StreamId.Guid, this.StreamId.Namespace,
-                    this.GenerateEvent(this.SequenceNumberCounter.Value), RequestContextExtensions.Export(this.serializationManager));
-
-                //set partition key
-                eventData.SetPartitionKey(this.StreamId.Guid.ToString());
-
-                //set offset
-                DateTime now = DateTime.UtcNow;
-                var offSet = this.StreamId.Guid.ToString() + now.ToString();
-                eventData.SetOffset(offSet);
-                //set sequence number
-                eventData.SetSequenceNumber(this.SequenceNumberCounter.Value);
-                //set enqueue time
-                eventData.SetEnqueuedTimeUtc(now);
-                eventDataList.Add(eventData);
-                this.logger.Info($"Generate data of SequemceNumber {SequenceNumberCounter.Value} for stream {this.StreamId.Namespace}-{this.StreamId.Guid}");
+
+                var eventData = EventHubBatchContainer.ToEventData<int>(
+                    this.serializationManager,
+                    this.StreamId.Guid,
+                    this.StreamId.Namespace,
+                    this.GenerateEvent(this.SequenceNumberCounter.Value),
+                    RequestContextExtensions.Export(this.serializationManager));
+
+                var wrapper = new WrappedEventData(
+                    eventData.Body,
+                    eventData.Properties,
+                    eventData.SystemProperties,
+                    partitionKey: this.StreamId.Guid.ToString(),
+                    offset: DateTime.UtcNow.Ticks,
```

Comment:
My main concern about this change is the offset. Is it a breaking change?

Reply:
Nop, it is behaving the same way as the previous one. In fact I haven't change the behavior at all. Just got the PR that is merged in master and adapted it to fit the old SDK on 3.x.